mage_ai/data_preparation/repo_manager.py

```diff
@@ -293,23 +293,28 @@ def get_variables_dir(
     return variables_dir
 
 
-project_uuid = None
-try:
-    with get_metadata_path() as f:
-        config = yml.load(f) or {}
-        project_uuid = config.get('project_uuid')
-except Exception:
-    pass
+def set_project_uuid_from_metadata():
+    global project_uuid
+    try:
+        with open(get_metadata_path(), 'r', encoding='utf-8') as f:
+            config = yml.load(f) or {}
+            project_uuid = config.get('project_uuid')
+    except Exception:
+        pass
```

Comment:
I wonder whether we should catch the exception. If the metadata file can't be loaded by yaml, the service shouldn't be started, right?

Reply:
Can you test the following cases?
* The metadata file doesn't exist
* The metadata file has in valid json
* It fails to open the metadata file randomly

Reply:
* i added a check for this, but the metadata file will be created in the `init_project_uuid` call
* this will cause the server to not start because the `init_project_uuid` call will fail
* not sure why this would happen, but it could cause problems... should i add a retry or something?